crates/testing/tests/tests_3/test_with_builder_failures.rs

```diff
@@ -0,0 +1,68 @@
+use std::time::Duration;
+
+use hotshot_example_types::node_types::{Libp2pImpl, MemoryImpl, PushCdnImpl, TestTypes};
+use hotshot_macros::cross_tests;
+use hotshot_testing::{
+    block_builder::SimpleBuilderImplementation,
+    test_builder::{BuilderChange, BuilderDescription, TestDescription},
+    txn_task::TxnTaskDescription,
+};
+
+// Test one node leaving the network.
+cross_tests!(
+    TestName: test_with_builder_failures,
+    Impls: [MemoryImpl, Libp2pImpl, PushCdnImpl],
+    Types: [TestTypes],
+    Ignore: false,
+    Metadata: {
+        let mut metadata = TestDescription::default_multiple_rounds();
+        // Every block should contain at least one transaction - builders are never offline
+        // simultaneously
+        metadata.overall_safety_properties.transaction_threshold = 1;
+        // Generate a lot of transactions so that freshly restarted builders still have
+        // transactions
+        metadata.txn_description = TxnTaskDescription::RoundRobinTimeBased(Duration::from_millis(1));
+
+        // Two builders running as follows:
+        // view 1st  2nd
+        // 0    Up   Down
+        // 1    Up   Up
+        // 2    Down Up
+        // 3    Up   Up
+        // 4    Up   Down
+        // 5    Up   Up
+        // 6    Down Up
+        // 7    Up   Up
+        // ...
+        //
+        // We give each builder a view of uptime before making it the only available builder so that it
+        // has time to initialize
+
+        // First builder will always respond with available blocks, but will periodically fail claim calls
+        #[allow(clippy::unnecessary_filter_map)] // False positive
```

Comment:
Is this a false positive? Does clippy not provide a working replacement option? It looks like you could fill the map then just swap the value for key 2 and 0 in this and the below builder without needing to map over it at all.

Reply:
> Is this a false positive?

Oh, wow. The reason I though it was a false positive was because this was supposed to be 
```rust
                0 => Some((view_num, BuilderChange::Down)),
                1 => Some((view_num, BuilderChange::Up)),
                _ => None, // We don't want to send Up when builder is up already, that'd
                                   // restart it
```
and I somehow convinced myself that that's what I wrote... Thank you for the catch!

> It looks like you could fill the map then just swap the value for key 2 and 0 in this and the below builder without needing to map over it at all.

Not sure what you mean by this though - the keys range from 0 to the number of views we're testing for, it's not only 0 and 2.

Reply:
Oh, hah pardon my blindness on the second point. This sounds fine.